Automattic/jetpack
# Add was_ecommerce_trial flag to site data from API

Related to Automattic/wc-calypso-bridge#1005

## Proposed changes:

* This PR adds the `was_ecommerce_trial` flag to the site data we return from the site REST APIs
  - This value will always be false for normal Jetpack sites, as the eCommerce trial is only available from WordPress.com

The primary use case for the new value is to make it possible for us to identify eCommerce trial sites where the trial has expired, as we want to limit user's abilities in Calypso when that happens. The related Calypso PR (which is still relatively exploratory from an implementation perspective) is Automattic/wp-calypso#74335.

### Other information:


## Jetpack product discussion



No real product changes.

## Does this pull request change what data or activity we track or use?



No changes.

## Testing instructions:






* Follow the instructions in [this comment](https://github.com/Automattic/jetpack/pull/29423#issuecomment-1464250837) to apply the changes in this PR to your WordPress.com development system
* Ensure that requests for `public-api.wordpress.com` are being routed to your dev system
* Open the [WordPress.com developer console](https://developer.wordpress.com/docs/api/console/)
* Test the following WP.COM API requests and verify that the `was_ecommerce_trial` flag is present in all cases, and is true for recently-created eCommerce trial sites, ideally including ones where the trial has since expired
    - To create a new trial site, visit wordpress.com/setup/wooexpress or follow the flow from woocommerce.com/start
  - `GET v1.1 /me/sites?fields=ID%2CURL%2Cname%2Cjetpack%2Cwas_ecommerce_trial` - the `fields` parameter is needed to avoid a huge response with all fields; if you have a lot of sites, restricting the fields will speed up the response _a lot_
  - `GET v1.1 /sites/:siteSlug` for current, expired, and non-trial site slugs
  - `GET v1.2 /me/sites?fields=ID%2CURL%2Cname%2Cjetpack%2Cwas_ecommerce_trial` - as above RE the `fields` parameter
  - `GET v1.2 /sites/:siteSlug` for current, expired, and non-trial site slugs
